src/schema/posts.ts

```diff
@@ -1101,6 +1219,44 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
   BaseContext
 >({
   Query: {
+    squadPostModerationsBySourceId: async (
+      _,
+      { sourceId }: { sourceId: string },
+      ctx: Context,
+      info,
+    ): Promise<GQLSquadPostModeration[]> => {
+      const isModerator = await isPrivilegedMember(ctx, sourceId);
+      if (!isModerator) throw new ForbiddenError('Access denied!');
+
+      return graphorm.query<GQLSquadPostModeration>(ctx, info, (builder) => ({
+        ...builder,
+        queryBuilder: builder.queryBuilder.where(
+          `"${builder.alias}"."sourceId" = :sourceId`,
+          { sourceId },
+        ),
+      }));
```

Comment:
Both users and mods can fetch this endpoint.
- If **user**: check if joined squad, then retrieve **my** pending posts
- if **admin/mod**: check permission, then return **all** pending posts

Reply:
I think the user getting their own moderation posts should be a different endpoint from the admin getting all of the ones for the squad